### PR TITLE
Add popup opener

### PR DESCRIPTION
**Pull request by CPatchane**
A component/wrapper which will open a link in a popup according to props provided.
https://cpatchane.github.io/cozy-ui/react/#popupopener

### CODE REVIEW

File: react/PopupOpener/index.jsx [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+  // Fixes dual-screen position
+  //                      Most browsers      Firefox
+  var dualScreenLeft = window.screenLeft || screen.left
+  var dualScreenTop = window.screenTop || screen.top
+
+  var width = window.innerWidth
+    ? window.innerWidth
+    : document.documentElement.clientWidth
+      ? document.documentElement.clientWidth
+      : screen.width
```
Comment (ptbrowne) [inline react/PopupOpener/index.jsx]: Not blocking : `window.innerWidth || document.documentElement.clientWidth || screen.width`

File: react/PopupOpener/index.jsx [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+  var height = window.innerHeight
+    ? window.innerHeight
+    : document.documentElement.clientHeight
+      ? document.documentElement.clientHeight
+      : screen.height
```
Comment (ptbrowne) [inline react/PopupOpener/index.jsx]: pareil

File: react/PopupOpener/index.jsx [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+    '',
+    title,
+    'scrollbars=yes, width=' +
+      w +
+      ', height=' +
+      h +
+      ', top=' +
+      top +
+      ', left=' +
+      left
```
Comment (ptbrowne) [inline react/PopupOpener/index.jsx]: Not blocking: would be clearer with a template string

File: react/PopupOpener/index.jsx [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+    newWindow.focus()
+  }
+  return newWindow
+}
+
+const PopupOpener = props => {
+  const { url, className, children, frameTitle, height, width } = props
+  return (
+    <a
+      onClick={() => popupCenter(url, frameTitle, width, height)}
```
Comment (ptbrowne) [inline react/PopupOpener/index.jsx]: Not blocking : `openCenteredPopup` would be clearer

File: react/PopupOpener/index.jsx [hunk 5]
```diff
@@ -0,0 +1,72 @@
...
+  url: PropTypes.string.isRequired,
+  className: PropTypes.string,
+  frameTitle: PropTypes.string,
+  height: PropTypes.number.isRequired,
+  width: PropTypes.number.isRequired
+}
+
+PopupOpener.defaultProps = {
+  className: ''
+}
```
Comment (ptbrowne) [inline react/PopupOpener/index.jsx]: No need to default the class name to nothing

File: react/PopupOpener/index.jsx [hunk 6]
```diff
@@ -0,0 +1,72 @@
...
+import PropTypes from 'prop-types'
+
+// customized function to center a popup window
+// source https://stackoverflow.com/a/16861050
+export function popupCenter(url, title, w, h) {
+  /* global screen */
+  // Fixes dual-screen position
+  //                      Most browsers      Firefox
+  var dualScreenLeft = window.screenLeft || screen.left
+  var dualScreenTop = window.screenTop || screen.top
```
Comment (ptbrowne) [inline react/PopupOpener/index.jsx]: Not blocking : screenLeft and screenTop seem more appropriate here.

**Comment by CPatchane** [comment]
Thanks @ptbrowne! Indeed, this copy/paste can greatly be improved, my bad.

File: react/PopupOpener/index.jsx (hunk 6 above)
Comment (CPatchane) [reply react/PopupOpener/index.jsx]: Removed to use directly `window.screenX` and `window.screenY`

**Pull request merged**